tendermint/tests/lite.rs

```diff
@@ -125,6 +197,57 @@ fn run_test_cases(cases: TestCases) {
     }
 }
 
+#[test]
+fn bisection_simple() {
+    let case: TestBisection =
+        serde_json::from_str(&read_json_fixture("many_header_bisection/happy_path")).unwrap();
+    run_bisection_test(case);
+}
+
+fn run_bisection_test(case: TestBisection) {
+    println!("{}", case.description);
+
+    let untrusted_height = case.height_to_verify.try_into().unwrap();
+    let trust_threshold = case.trust_level;
+    let trusting_period = case.trust_options.period;
+    let now = case.now;
+
+    let provider = case.primary;
+    let req = MockRequester::new(provider.chain_id, provider.lite_blocks);
+
+    let expected_output = case.expected_output;
+
+    let trusted_height = case.trust_options.height.try_into().unwrap();
+    let trusted_header = &req
+        .signed_header(trusted_height)
+        .expect("could not 'request' signed header");
+    let trusted_vals = &req
+        .validator_set(trusted_height + 1)
+        .expect("could not 'request' validator set");
+
+    let trusted_state = TrustedState::new(trusted_header, trusted_vals);
+
+    let output: String;
+
+    match lite::verify_bisection(
+        trusted_state,
+        untrusted_height,
+        trust_threshold,
+        &trusting_period.into(),
+        &now.into(),
+        &req,
+    ) {
+        Ok(_) => {
+            // TODO: should we make some assertions on the returned new_states?
```

Comment:
Yes, we can do that. My only concern is that since this will test each step of bisection, any changes in the bisection algorithm in future might require updating each test file every time.

Reply:
Fair enough. Then maybe we do a bunch of tests pass/fail where we check that it synced to the final height or not, and then a few tests that also check the intermediate headers, and only those will have to be updated with if the bisection algorithm changes (though at first thought I don't think we're expecting it to change :) )